actionners/kubernetes/sysdig/sysdig.go

```diff
@@ -0,0 +1,292 @@
+		Image:      defaultImage,
+	}
+}
+
+func (a Actionner) Checks(event *events.Event, _ *rules.Action) error {
+	if event.Source == "k8s_audit" {
+		event.OutputFields["k8s.ns.name"] = event.OutputFields["ka.target.namespace"]
+		if event.OutputFields["ka.target.pod.name"] != "" {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.pod.name"]
+		} else {
+			event.OutputFields["k8s.pod.name"] = event.OutputFields["ka.target.name"]
+		}
+	}
+	return k8sChecks.CheckPodExist(event)
+}
+
+func (a Actionner) Run(event *events.Event, action *rules.Action) (utils.LogLine, *models.Data, error) {
+	podName := event.GetPodName()
+	namespace := event.GetNamespaceName()
+
+	objects := map[string]string{
+		"pod":       podName,
+		"namespace": namespace,
+	}
+
+	var parameters Parameters
+	err := utils.DecodeParams(action.GetParameters(), &parameters)
+	if err != nil {
+		return utils.LogLine{
+			Objects: nil,
+			Error:   err.Error(),
+			Status:  utils.FailureStr,
+		}, nil, err
+	}
+
+	if parameters.Duration == 0 {
+		parameters.Duration = defaultDuration
+	}
+
+	if parameters.Duration > 30 {
+		parameters.Duration = defaultMaxDuration
+	}
+
+	if parameters.Image == "" {
+		parameters.Image = defaultImage
+	}
+
+	if parameters.Scope == "" {
+		parameters.Scope = defaultScope
+	}
+
+	client := k8s.GetClient()
+
+	pod, _ := client.GetPod(podName, namespace)
+	containers := k8s.GetContainers(pod)
+	if len(containers) == 0 {
+		err = fmt.Errorf("no container found")
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  utils.FailureStr,
+		}, nil, err
+	}
+
+	job, err := client.CreateJob("falco-talon-sysdig", namespace, parameters.Image, pod.Spec.NodeName, defaultTTL)
+	if err != nil {
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  utils.FailureStr,
+		}, nil, err
+	}
+
+	objects["job"] = job
+
+	timeout := time.NewTimer(20 * time.Second)
+	ticker := time.NewTicker(300 * time.Millisecond)
+	defer timeout.Stop()
+	defer ticker.Stop()
+
+	var ready bool
+	var jPod, jContainer string
+	for !ready {
+		select {
+		case <-timeout.C:
+			err = fmt.Errorf("the job '%v' in the namespace '%v' for the sysdig capture is not ready", job, namespace)
+			return utils.LogLine{
+				Objects: objects,
+				Error:   err.Error(),
+				Status:  utils.FailureStr,
+			}, nil, err
+		case <-ticker.C:
+			p, err2 := client.ListPods(metav1.ListOptions{LabelSelector: "batch.kubernetes.io/job-name=" + job})
+			if err2 != nil {
+				return utils.LogLine{
+					Objects: objects,
+					Error:   err2.Error(),
+					Status:  utils.FailureStr,
+				}, nil, err2
+			}
+			if len(p.Items) > 0 {
+				if p.Items[0].Status.Phase == corev1.PodRunning && p.Items[0].Status.ContainerStatuses[0].Ready {
+					jPod = p.Items[0].Name
+					jContainer = p.Items[0].Spec.Containers[0].Name
+					ready = true
+				}
+			}
+		}
+	}
+
+	command := []string{"tee", "/tmp/talon-script.sh", "/dev/null"}
```

Comment:
I see now why you're using your own image, is this something we want to have in official sysdig image?

Reply:
never considered it